Kitodo/src/main/java/org/kitodo/production/helper/metadata/MediaPartialHelper.java

```diff
@@ -0,0 +1,148 @@
+
+import org.apache.commons.lang3.tuple.Pair;
+import org.kitodo.api.dataformat.LogicalDivision;
+import org.kitodo.api.dataformat.MediaPartial;
+import org.kitodo.api.dataformat.PhysicalDivision;
+import org.kitodo.api.dataformat.View;
+import org.kitodo.api.dataformat.Workpiece;
+import org.kitodo.production.metadata.MetadataEditor;
+
+public class MediaPartialHelper {
+
+    public static final Pattern FORMATTED_TIME_PATTERN = Pattern.compile("([0-1]\\d|2[0-3]):[0-5]\\d:[0-5]\\d\\.\\d{3}");
+
+    private static final MediaPartialLogicalDivisionComparator logicalDivisionComparator = new MediaPartialLogicalDivisionComparator();
+
+    /**
+     * Convert formatted time to milliseconds.
+     *
+     * @param formattedTime
+     *         The formatted time in form of {@value #FORMATTED_TIME_PATTERN}
+     * @return The milliseconds
+     */
+    public static Long convertFormattedTimeToMilliseconds(String formattedTime) {
+        return Duration.between(LocalTime.MIN, LocalTime.parse(formattedTime)).toMillis();
+    }
+
+    /**
+     * Convert milliseconds to formatted time.
+     *
+     * @param milliseconds
+     *         The milliseconds
+     * @return The formatted time
+     */
+    public static String convertMillisecondsToFormattedTime(Long milliseconds) {
+        Duration duration = Duration.ofMillis(milliseconds);
+        return String.format("%02d:%02d:%02d.%03d", duration.toHours(), duration.toMinutesPart(),
+                duration.toSecondsPart(), duration.toMillisPart());
+    }
+
+    /**
+     * Calculates the extent field of every media partial and sort media partials by begin.
+     *
+     * @param logicalDivisions
+     *         The logical divisions of media partials.
+     * @param mediaDuration
+     *         The media duration.
+     */
+    public static void calculateExtentAndSortMediaPartials(List<LogicalDivision> logicalDivisions, Long mediaDuration) {
+        calculateExtentForMediaPartials(logicalDivisions, mediaDuration);
+        logicalDivisions.sort(logicalDivisionComparator);
+    }
+
+    /**
+     * Calculate the extent of a media partial.
+     *
+     * <p>
+     * Calculates the duration or extent of a media partial until the next one or until the end of the media.
+     *
+     * @param logicalDivisions
+     *         The logical divisions of media partials.
+     * @param mediaDuration
+     *         The media duration.
+     */
+    private static void calculateExtentForMediaPartials(List<LogicalDivision> logicalDivisions, Long mediaDuration) {
+        // sorting reverse by begin
+        logicalDivisions.sort(logicalDivisionComparator.reversed());
+
+        ListIterator<LogicalDivision> iterator = logicalDivisions.listIterator();
+        LogicalDivision previousLogicalDivision = null;
+        while (iterator.hasNext()) {
+            LogicalDivision logicalDivision = iterator.next();
+            MediaPartial mediaPartial = logicalDivision.getViews().getFirst().getPhysicalDivision()
```

Comment:
Don't we have to check whether the logical division _has_ views before retrieving the first? This method is called by a public method, which means anyone could use it in the future and pass a logical division without checking whether it has views or not.